### PR TITLE
fix: keep bool parameters as bool in sklearn_tuner (#108)

### LINKED ISSUE #108: sklearn_tuner change Bool to Int
I got the error "The 'bootstrap' parameter of cross_val_predict must be an instance of 'bool' or an instance of 'numpy.bool_'. Got 0 instead." when running

`sklearn_tuner(RandomForestRegressor, space_cfg, X, y, metric = r2_score, max_iter = 16)`

The sklearn version is 1.6.1. I believe it may be because in line 35 of bool_param.py, the bool variable is treated as numeric, and then in line 79 of `sklearn_tuner.py` it changes the bool to int.

### PR DESCRIPTION
**Pull request by Nas01010101**
## Summary

`sklearn_tuner` int-casts every discrete numeric hyper-parameter:

```python
if space.paras[k].is_numeric and space.paras[k].is_discrete:
    hyp[k] = int(hyp[k])
```

`BoolPara` reports **both** `is_numeric` and `is_discrete` as `True`, so a boolean hyper-parameter (e.g. a `RandomForest`'s `bootstrap`) is cast to `0`/`1`. Modern scikit-learn then rejects it:

> The 'bootstrap' parameter ... must be an instance of 'bool' ... Got 0 instead.

This is issue #108.

## Fix

Special-case `BoolPara` so boolean parameters stay python `bool`, and only int-cast the remaining (non-bool) discrete numerics.

## Tests

Adds `test/test_sklearn_tuner_bool.py` — a recorder estimator that asserts the parameter *type* it receives at `fit`. Fails before the fix (`bool` coerced to `int`), passes after.

```
1 passed
```

Closes #108.

### CODE REVIEW

**Comment by Nas01010101** [comment]
Verified this still applies cleanly against current `master` (0 commits behind) and the added test still fails on revert.

Heads up for merge order: this PR and #123 both touch the `for k in hyp:` loop body in `sklearn_tuner.py` (this one changes the cast logic, #123 wraps the loop in `warnings.catch_warnings()`), so merging both will produce a conflict on whichever lands second. Happy to rebase the other one once you've merged one of them — just ping the branch.
